apps/main_app/lib/main_app.ex

```diff
@@ -3,22 +3,37 @@ defmodule MainApp do
 
   use Application
 
-  @subscriber Application.compile_env(:main_app, :subscriber, nil)
-  @replicator Application.compile_env(:main_app, :replicator, nil)
-  @publisher Application.compile_env(:main_app, :publisher, nil)
+  @subscribers Application.compile_env(:main_app, :subscribers, [])
+  @publishers Application.compile_env(:main_app, :publishers, [])
 
   @impl true
   def start(_type, _args) do
     children = [
-      @subscriber,
-      {@replicator,
-       [host: "localhost", database: "postgres", username: "postgres", password: "postgres"]},
-      {@publisher, "/tmp/replication.log"},
+      Hub,
+      get_children_spec(@publishers),
+      get_children_spec(@subscribers)
     ]
 
+    children = List.flatten(children)
+
     # See https://hexdocs.pm/elixir/Supervisor.html
     # for other strategies and supported options
     opts = [strategy: :one_for_one, name: MainApp.Supervisor]
     Supervisor.start_link(children, opts)
   end
+
+  defp get_children_spec(config_properties) do
+    Enum.map(config_properties, fn property ->
+      case property do
+        module_name when is_atom(module_name) ->
```

Comment:
Hmm, I wanted `name` to be optional. Now I have 2 options: either I fill in everything, or only the module name. How do I define a single `PgSubscriber` (without `name` parameter) when it always needs database connection parameters?

Reply:
Yeah making name optional makes more sense... Since init args where hardcoded I completely forgot about them. :smile:

Reply:
I have changed so the `name` is actually never required. Thanks `Registry`. :smile: